### PR TITLE
feat(pipelines): build tagging

### PR DESCRIPTION
**Pull request by c3ppc3pp**
This PR adds tagging capabilities to the pipeline:
when a git tag is set and the build is running against that tag, it will create a docker image with the same tag.

Global changes:
- pipeline_set_output_tag.sh now will output version too, which is 0.0.0 by default, will be set to the tag if available. Docker Image tag will be the commit checksum, unless its built against develop or main branches, then the image tag will be latest and stable respectively.

With crux/crux-ui pipeline:
- Check if package.json version is the same as the tag, if not, build will fail.
- Docker Images are built from the dockerfiles, we wont use any artifact with these.
- Docker will tag additionally if necessary(we can apply multiple tags automatically with a bit of work from now on), then push the images. Same thing goes for signing.

crux/crux-ui build:
- Timezone now can be set during runtime instead of buildtime via TZ environment variable.
- Removed npx from crux build processes
- Fixed some missing deps in the crux build processes

### CODE REVIEW

**Comment by c3ppc3pp** [comment]
@m8vago Please test the images produced by the new build process, especially the recaptcha things! Thank you! 

CRUX-UI https://github.com/dyrector-io/dyrectorio/pkgs/container/dyrectorio%2Fweb%2Fcrux-ui/29902123?tag=0.0.1

CRUX https://github.com/dyrector-io/dyrectorio/pkgs/container/dyrectorio%2Fweb%2Fcrux/29902106?tag=0.0.1

File: web/crux-ui/Dockerfile [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
+COPY --from=BUILDER /app/package.json ./package.json
+COPY --from=BUILDER --chown=node:node /app/.next ./.next
+COPY --from=BUILDER /app/node_modules ./node_modules
 
 USER node
 
 EXPOSE 3000
+ENV PORT 3000
 
 ENTRYPOINT [ "npm", "run", "prod"]
```
Comment (polaroi8d) [inline web/crux-ui/Dockerfile:39]: Could you add NL at EOF, please?

File: .github/workflows/web-crux-ui.yml [hunk 2]
```diff
@@ -73,18 +62,20 @@ jobs:
...
     needs: check
     container: docker:latest
-    if: github.ref_name == 'develop' || github.ref_name == 'main'
+    if: github.ref_name == 'develop' || github.ref_name == 'main' || github.ref_type == 'tag'
     defaults:
       run:
         working-directory: ${{ env.WORKING_DIRECTORY }}
     steps:
 
-    - name: Download artifact
```
Comment (polaroi8d) [inline .github/workflows/web-crux-ui.yml:82]: We are in the same 'context', this is the reason why we don't need to download the artifacts?

**Review by polaroi8d — CHANGES_REQUESTED** [review]

File: .github/workflows/web-crux-ui.yml (hunk 2 above)
Comment (c3ppc3pp) [reply .github/workflows/web-crux-ui.yml:82]: Dockerfile does a prod build too, so downloading the artifacts are not necessary.

File: web/crux/package.json [hunk 3]
```diff
@@ -33,7 +33,8 @@
     "database:up": "docker-compose -f docker-compose.database.yaml up -d",
     "database:down": "docker-compose -f docker-compose.database.yaml down",
     "test:crux": "npm run docker:up && npx prisma migrate deploy && jest -i && npm docker:down",
-    "compodoc": "npx compodoc -p tsconfig.doc.json --theme stripe"
+    "prismagen": "prisma generate",
+    "compodoc": "compodoc -p tsconfig.doc.json --theme stripe"
```
[the remark this answers is not in the document]
Comment (c3ppc3pp) [reply web/crux/package.json:37]: The "compodoc" command was there already, I assumed we are using it. If it is unused we can remove it no issues.

File: web/crux/crux-docker-entrypoint.sh [hunk 4]
```diff
@@ -13,6 +13,9 @@ generate_cert() {
     fi
 }
 
+# ln -sf /usr/share/zoneinfo/${TIME_ZONE} /etc/localtime
```
[the remark this answers is not in the document]
Comment (c3ppc3pp) [reply web/crux/crux-docker-entrypoint.sh]: Good catch! Fixed.

File: web/crux-ui/Dockerfile (hunk 1 above)
Comment (c3ppc3pp) [reply web/crux-ui/Dockerfile]: Yes, done.

**Review by polaroi8d — APPROVED** [review]
LGTM, thanks.

File: .github/workflows/web-crux.yml [hunk 5]
```diff
@@ -76,19 +70,19 @@ jobs:
...
     steps:
     - name: Checkout
       uses: actions/checkout@v3
 
-    - name: Download /dist artifact
-      uses: actions/download-artifact@v3
-      with:
-        name: crux-web-dist
-        path: ${{ env.WORKING_DIRECTORY }}/dist
+      # if tag isn't the version setted in package.json will make the job fail
```
Comment (robot9706) [inline .github/workflows/web-crux.yml:81]: `set` instead of `setted` ?

Comment (c3ppc3pp) [reply .github/workflows/web-crux.yml]: You're right! Thanks! Also fixed in another pipeline where I used this comment.

**Review by robot9706 — APPROVED** [review]
😎

**Comment by c3ppc3pp** [comment]
Hold it off for a moment. There's a showstopper bug in cert-generation script.

**Comment by c3ppc3pp** [comment]
Re-review it please!

File: .github/workflows/web-crux.yml [hunk 6]
```diff
@@ -97,12 +91,21 @@ jobs:
...
 
-    - name: Docker build and push
+    - name: Docker build 
       run: |
-        docker-compose build
-        docker-compose push crux
+        docker build . -t ${REGISTRY}/${IMAGE_NAME}:${{ needs.prerequirements.outputs.tag }}
       env:
-        CRUX_IMAGE_TAG: ${{ needs.prerequirements.outputs.tag }}
+        CRUX_UI_IMAGE_TAG: ${{ needs.prerequirements.outputs.tag }}
```
Comment (m8vago) [inline .github/workflows/web-crux.yml:98]: ui?

Comment (m8vago) [reply .github/workflows/web-crux.yml:98]: they have a shared image tag now?

**Pull request merged**